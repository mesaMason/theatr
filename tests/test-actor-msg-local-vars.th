dolphin(int weight, int age):
    print("inside dolphin.")
    receive:
        swim():
            int s = 1
            print("swim()'s local var:")
            print(s)
        eat():
            int e = 2
            print("eat()'s local var:")
            print(e)
    drop:
        int d = 0
        print("drop()'s local var:")
        print(d)
    after:
        return;
        
func main() -> int:
    int weight = 10
    int age = 3
    actor d = new dolphin(weight, age)
    asdfasodfasdf() | d
    eat() | d
    swim() | d
    die() | d
    return 0

