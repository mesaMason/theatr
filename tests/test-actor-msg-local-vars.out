new dolphin created.
inside dolphin.
drop()'s local var:
0
eat()'s local var:
2
swim()'s local var:
1        
